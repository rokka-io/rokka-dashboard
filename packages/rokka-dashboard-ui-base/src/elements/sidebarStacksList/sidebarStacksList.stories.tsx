import { boolean, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { colors } from '../../identity/colors/colors';
import { SidebarStacksList } from './sidebarStacksList';

storiesOf('Elements / Sidebar', module)
  .addDecorator(story => (
    <div
      style={{
        position: 'relative',
        background: colors.gray.sidebar,
        width: '200px',
        height: '100%',
        color: colors.tints.white
      }}
    >
      {story()}
    </div>
  ))
  .add('SidebarStacksList', () => (
    <SidebarStacksList style={{ border: '3px dashed red' }}>{text('text', 'Some text')}</SidebarStacksList>
  ));
