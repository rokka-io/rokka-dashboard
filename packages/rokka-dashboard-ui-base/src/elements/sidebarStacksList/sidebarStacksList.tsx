import styled from 'styled-components';

export const SidebarStacksList = styled.div`
  overflow-x: hidden;
  overflow-y: auto;
  height: calc(100vh - 310px);
`;
