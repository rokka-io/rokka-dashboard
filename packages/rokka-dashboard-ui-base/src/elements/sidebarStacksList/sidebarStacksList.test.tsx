import { shallow } from 'enzyme';
import React from 'react';
import { SidebarStacksList } from './sidebarStacksList';

describe('Elements', () => {
  describe('Sidebar', () => {
    it('renders a SidebarStacksList correctly.', () => {
      const component = shallow(<SidebarStacksList />);

      expect(component).toMatchSnapshot();
    });
  });
});
