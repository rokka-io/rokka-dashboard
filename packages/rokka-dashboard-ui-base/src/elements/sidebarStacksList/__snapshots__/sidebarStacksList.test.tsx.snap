// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Sidebar renders a SidebarStacksList correctly. 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  overflow-x: hidden;
  overflow-y: auto;
  height: calc(100vh - 310px);
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
